source/common/router/retry_state_impl.cc

```diff
@@ -240,6 +243,20 @@ RetryStatus RetryStateImpl::shouldRetry(bool would_retry, DoRetryCallback callba
   return RetryStatus::Yes;
 }
 
+bool RetryStateImpl::retryBudgetExceeded() {
+  ASSERT(retry_budget_);
+
+  const uint64_t current_active = cluster_.resourceManager(priority_).requests().count() +
```

Comment:
Does this work for HTTP/1.1? IIRC we have a not great situation in which HTTP/1.1 tracks connections and HTTP/2 tracks requests. IMO we should fix the HTTP/1.1 to also track requests even if we don't break on them so that this logic is consistent? If that's the case can you do that in a different PR?

Reply:
My initial way of calculating this (by summing connections/requests/active requests) was an attempt to account for this. In another PR, are you suggesting we keep a counter of active requests that works for all HTTP versions?

Reply:
Basically I'm just suggesting to inc/dec requests on the resource manager inside the HTTP/1.1 connection pool, even if we don't circuit break on it. This should be an easy/safe change that would then make it consistent with HTTP/2?